src/Fixer/Whitespace/TypehintSpaceFixer.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\AllowedValueSubset;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\TypeAnalysis;
+use PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ * @author John Paul E. Balandan, CPA <paulbalandan@gmail.com>
+ */
+final class TypehintSpaceFixer extends AbstractFixer implements ConfigurableFixerInterface
```

Comment:
Can we go with the name `TypeSpaceFixer`? Most of the fixers are using a convention of types, not typehints. With this rename/deprecation, it would leave only 1 fixer with the name "typehint".

Reply:
Would that be confusing with the TypesSpacesFixer?

Reply:
Ha, missed that 😄. That would be very confusing.

Maybe then one fixer for all the types relates spaces? Would it be better to have spaces for union|itnesection and space after a type as separate fixers or not?

Reply:
I think we can adapt `TypesSpacesFixer` to accommodate spaces after the type. The fixer name itself seems not restrictive of what spaces it can fix.